build.gradle

```diff
@@ -139,7 +139,59 @@ allprojects {
     }
 }
 
-task tar(type: Tar) {
+tasks.register('dist') {
+    subprojects.forEach { subProject ->
+        dependsOn("${subProject.path}:jar")
+    }
+    def includedProjects =
+            ["eventmesh-common",
+             "eventmesh-meta:eventmesh-meta-api",
+             "eventmesh-metrics-plugin:eventmesh-metrics-api",
+             "eventmesh-protocol-plugin:eventmesh-protocol-api",
+             "eventmesh-retry:eventmesh-retry-api",
+             "eventmesh-runtime",
+             "eventmesh-security-plugin:eventmesh-security-api",
+             "eventmesh-spi",
+             "eventmesh-starter",
+             "eventmesh-storage-plugin:eventmesh-storage-api",
+             "eventmesh-trace-plugin:eventmesh-trace-api",
+             "eventmesh-webhook:eventmesh-webhook-api",
+             "eventmesh-webhook:eventmesh-webhook-admin",
+             "eventmesh-webhook:eventmesh-webhook-receive"]
+    outputs.dirs('dist/apps', 'dist/bin', 'dist/conf', 'dist/lib', 'dist/licenses')
+    doLast {
+        includedProjects.each {
+            def project = findProject(it)
```

Comment:
Better name it `subProject` to indicate that all `from project...` equals to `${projectDir}/...` and all `into 'dist/xxx'` equals to `${rootDir}/dist/xxx`.

---

Maybe `${projectDir}` and `${rootDir}` will be useful to indicate the relative and absolute path, because I spent 5 mins to prove the change equals to the old one. 😁 But the current writting do looks like more concise, so it depends on you.